### PR TITLE
Remove redundant clientside filtering from iterMessages method

### PR DESCRIPTION
**Pull request by KarKarbI4**
Hi all!
## PR Motivation
TL;DR;
I suggest to remove client side filtering by formId from iterMesssages, because it works on server side now

Two more benefits:
1. Solves issue when chunk contains 0 messages matching fromId, but 100 from another user (example down below)
2. Doesnt remove formId from Api.messages.Search query, which makes loading faster

----------
## Initial problem:
I've run into an issue with client.iterMessages

When trying to get messages with search by me in chat with user, I get an exception
```js
client.iterMessages(dialog.entity, {
      search: keyword,
      fromUser: 'me'
    });
```

```
node_modules/.pnpm/telegram@2.6.13/node_modules/telegram/client/messages.j3
    this.request.offsetId = Number(lastMessage.id);
                                               ^

TypeError: Cannot read properties of undefined (reading 'id')
    at _MessagesIter._updateOffset (/home/viktor/dev/tgMessageCleaner/node_modules/.pnpm/telegram@2.6.13/nod)
    at _MessagesIter._loadNextChunk (/home/viktor/dev/tgMessageCleaner/node_modules/.pnpm/telegram@2.6.13/no)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Object.next (/home/viktor/dev/tgMessageCleaner/node_modules/.pnpm/telegram@2.6.13/node_modules/)
    at async findMessagesByKeyword (file:///home/viktor/dev/tgMessageCleaner/index.mjs:83:22)
    at async file:///home/viktor/dev/tgMessageCleaner/index.mjs:23:18
 ELIFECYCLE  Command failed with exit code 1.
```

I debugged, that's happing because there are more than 100 messages from another user matching search, but 0 from me and library-side filtering fails

I guess that library-side filtering is present, because telegram used to ignore fromId parameter in Search when entity is User chat. But now server side filtering by fromId works well in User chats too

To check server side filtering, I've tried:
```js
const result = await client.invoke(new Api.messages.Search({
  peer: "partnerusername",
  q: 'some word you partner likes but you do not use ever',
  filter: new Api.InputMessagesFilterEmpty(),
  limit: 50,
  fromId: "your user name"
}));
```
returns 0 messages for me with formId parameter specified and some messages without it (its messages from another user)

So server side filtering works fine now

### CODE REVIEW

**Review by painor — APPROVED** [review]
Looks good to me, will have to test it a bit before merge
